pkg/controller/grafanadashboard/dashboard_controller.go

```diff
@@ -180,6 +184,11 @@ func (r *ReconcileGrafanaDashboard) importDashboard(d *i8ly.GrafanaDashboard) (r
 		return reconcile.Result{}, defaultErrors.New("operator namespace not yet known")
 	}
 
+	json, err := r.loadDashboardFromURL(d)
+	if err == nil {
+		d.Spec.Json = json
```

Comment:
This will overwrite the json field when the dashboard is updated. I wonder if it would be better to keep that json always as a fallback option when resolving the URL fails instead of overwriting it with the last downloaded version? What do you think?

Reply:
I agree and that was the idea. If the URL fails to resolve it won't overwrite the json field as it won't satisfy the if statement. not sure whether I am right in saying that.